### PR TITLE
Program for matrix multiplication of 2-D array in c

### PR DESCRIPTION
**Pull request by shubh045**
<hr>

## Description 📜

Explained about array and matrix multiplication and write a program to multiply two matrices #7095

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+## Matrix multiplication
+If A is a m x n matrix and B is a n x p matrix <br>
+![matrix](./assets/img1.png)
+
+then multiplication of A and B, let say C, will be <br>
+![matrix](./assets/img2.png) <br>
+Here, C is a m x p matrix. To multiply two matrices, the column of first matrix should be equal to the row of second matrix.
+
+## Algorihtm for matrix multiplication of 2-D array
```
Comment (vasu-1) [inline C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md:22]:
```suggestion
## Algorithm for matrix multiplication of 2-D array
```

File: C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+1. if column of 1st matrix is not equal to row of second matrix
+
+```
+Enter no. of rows of 1st matrix: 2
+Enter no. of columns of 1st matrix: 2
+Enter no. of rows of 2nd matrix: 3
+Enter no. of column of 2nd matrix: 2
+
+It is not possible to multipltly these two matrices,
```
Comment (vasu-1) [inline C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md:116]:
```suggestion
It is not possible to multiply these two matrices,
```

File: C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    scanf("%d", &c1);
+    printf("Enter no. of rows of 2nd matrix: ");
+    scanf("%d", &r2);
+    printf("Enter no. of column of 2nd matrix: ");
+    scanf("%d", &c2);
+
+    // conditional statments
+    if (c1 != r2) //if number of columns of 1st matrix is not equal to no. of rows of 2nd matrix
+    {
+        printf("\nIt is not possible to multipltly these two matrices,\n");
```
Comment (vasu-1) [inline C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md:58]:
```suggestion
        printf("\nIt is not possible to multiply these two matrices,\n");
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Binary files are not allowed.!
Just copy and paste your photo into readme ! file !
And do suggested changes !

File: C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md (hunk 1 above)
Comment (deeqakkk) [inline C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md:22]: shubham, fix this typo "Algorihtm"

**Review by deeqakkk — CHANGES_REQUESTED** [review]
Your content has a bit of plagiarism, do mention websites(you have referred from) in the reference section.

**Comment by shubh045** [comment]
I have made the suggested changes, please review it.

File: C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+7 8
+Multplication of m1 and m2 is: 
+19 22
+43 50
+```
+
+
+# References
+
+https://en.wikipedia.org/wiki/Matrix_multiplication
```
Comment (deeqakkk) [inline C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md:143]: add links like this [Google](www.google.com)
website

**Review by deeqakkk — CHANGES_REQUESTED** [review]
Make this last change, the rest of the things are good to go.

File: C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md (hunk 4 above)
Comment (shubh045) [reply C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md]: Changes done, please review it

**Review by deeqakkk — APPROVED** [review]
@Suvraneel its good to be merged

**Comment by Suvraneel** [comment]
@shubh045  LGTM ❤️  Good work :octocat: 

(this would have been complete if Strassen's Matrix multiplication method was discussed as well...
But okay that is a big one, it can be a contribution for another day 😉 )

**Pull request merged**